model/src/main/scala/za/co/absa/atum/model/utils/JsonSyntaxExtensions.scala

```diff
@@ -36,16 +36,20 @@ object JsonSyntaxExtensions {
 
   implicit class JsonDeserializationSyntax(jsonStr: String) {
     def as[T: Decoder]: T = {
-      decode[T](jsonStr) match {
+      asSafe[T] match {
         case Right(value) => value
-        case Left(error) => throw new RuntimeException(s"Failed to decode JSON: $error")
+        case Left(error) => throw error
       }
     }
 
+    def asSafe[T: Decoder]: Either[io.circe.Error, T] = {
```

Comment:
You could make this method private.

Reply:
I think I originally used it in Reader, but seems dropped it since. Will make it private 👍

Reply:
I have the suspicion that it will be used in actual Reader successors - and therefore turned back to public. If not I will remove it in that future PR.